util/oidc/oidc.go

```diff
@@ -367,12 +387,49 @@ func (a *ClientApp) HandleCallback(w http.ResponseWriter, r *http.Request) {
 	if a.secureCookie {
 		flags = append(flags, "Secure")
 	}
-	var claims jwt.MapClaims
+	claims := Claims{}
 	err = idToken.Claims(&claims)
 	if err != nil {
 		http.Error(w, err.Error(), http.StatusInternalServerError)
 		return
 	}
+	groups := claims.Groups
+
+	// Some SSO implementations (Okta) require a call to
+	// the OIDC user info path to get attributes like groups
+	if a.userInfoPath != "" {
+		userInfo, err := a.GetUserInfo(token.AccessToken, a.settings.IssuerURL(), a.userInfoPath)
+		if err != nil {
+			w.WriteHeader(401)
+			_, _ = w.Write([]byte(fmt.Sprintf("failed to get groups claim: %v", err)))
+			return
+		}
+		if idToken.Subject != userInfo.Subject {
+			log.Warning("Subject of Claims from UserInfo endpoint didn't match Subject of idToken, drop usage of the UserInfo endpoint data")
+			return
+		}
+		groups = userInfo.Groups
+	}
+
+	// reassamble the JWT claims
+	argoClaims := Claims{
+		groups,
+		jwt.RegisteredClaims{
+			ExpiresAt: jwt.NewNumericDate(idToken.Expiry),
+			IssuedAt:  jwt.NewNumericDate(time.Now()),
+			NotBefore: jwt.NewNumericDate(time.Now()),
+			Issuer:    "argocd", // token will now be validated against argocd instead of IDP
```

Comment:
Can this instead reference the constant, `SessionManagerClaimsIssuer`?

Reply:
Sure, I references the const in the `session` package.